### PR TITLE
chore: better error handling on search screen

### PR DESCRIPTION
**Pull request by MounirDhahri**
### Description

This PR brings some improvement to the way we handle errors in the search screen inspired by @gkartalis comments on [Slack](https://artsy.slack.com/archives/C05EQL4R5N0/p1695978699044759). This includes:
- Using the new middleware for the search screen - this means that failure in the different rails, won't lead to the search tab failing
- Use better error messages when the search query fails 
- Only show the retry button once to retry the query, the goal behind that is to avoid having the user blocked in infinite cycle of refreshing

https://github.com/artsy/eigen/assets/11945712/6a49dc41-18f9-4b04-b1ed-7d76344bbf41

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- better error handling on search screen - george, mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (better error handling on search screen - george, mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against be64292ae9203929a981618674d84f132b65d049
</p>

**Review by gkartalis — APPROVED** [review]
👏

File: src/app/Components/ArtistAutosuggest/ArtistAutosuggestResults.tsx [hunk 1]
```diff
@@ -86,7 +86,7 @@ const ArtistAutosuggestResultsFlatList: React.FC<{
       <AutoSuggestBoxContainer>
         <Spacer y={4} />
         <Text variant="sm-display" textAlign="center">
-          Sorry, we couldn’t find anything for {quoteLeft}
+          Sorry, we couldn't find anything for {quoteLeft}
```
Comment (dblandin) [inline src/app/Components/ArtistAutosuggest/ArtistAutosuggestResults.tsx:89]: **question / non-blocking**: This seems a bit tangential to the goal of this PR. Could we omit out these stylistic typography changes for now? I'd like to confirm with design how we prefer to format our apostrophes. For example, in palette-mobile, [it seems like we prefer the curved style](https://github.com/artsy/palette-mobile/blob/6326befa707d08fe99a2508e105b2e7e07d464ff/src/utils/text.ts#L14).

Comment (MounirDhahri) [reply src/app/Components/ArtistAutosuggest/ArtistAutosuggestResults.tsx:89]: yes of course, I will do that.

Comment (MounirDhahri) [reply src/app/Components/ArtistAutosuggest/ArtistAutosuggestResults.tsx:89]: I found both patterns in Eigen - will update them

Comment (MounirDhahri) [reply src/app/Components/ArtistAutosuggest/ArtistAutosuggestResults.tsx]: rebased and dropped the commit.

**Review by dblandin — APPROVED** [review]
🙏

**Pull request merged**